packages/venia-ui/lib/i18n/en_US.json

```diff
@@ -34,8 +34,7 @@
     "Product Suggestions": "Product Suggestions",
     "Register": "Register",
     "Returns": "Returns",
-    "searchPage.searchTermLabel": "Showing results for",
-    "searchPage.searchTermValue": "{term} in {category}",
+    "searchPage.searchTerm": "Showing results for <highlight>{term}</highlight>{category, select, null {} other {<highlight> in {category}</highlight>}}:",
```

Comment:
Well this one was fun 😅 .

It's either displaying "Showing results for {searchTerm}" or "Showing results for {searchTerm} in {category}".

Additionally, the "{searchTerm}" / "{searchTerm} in {category}" text should be emphasized (bolded) on the page.  See this definition and the accompanying `searchPage.searchTerm` `FormattedMessage` in `searchPage.js` on how to accomplish both of these at the same time.

Note that some languages might have the emphasized text / search term elsewhere in the text, so we can't just translate "Search results for" separately.

Reply:
Also, nitpicking, but should `in` be highlighted?

![](https://i.gyazo.com/bfedb913be021550209ec67883391d77.png)

[![Image from Gyazo](https://i.gyazo.com/8851d6e9b3bd6cc6060169e7da21a645.png)](https://gyazo.com/8851d6e9b3bd6cc6060169e7da21a645)

Reply:
> wouldn't it have been easier to just conditionally render the strings based on category truthiness?

I actually did this initially, lol. I had two separate translations, one for "Showing results for" and then one for "{term} in {category}".

But I don't think that is the right way to do it because some languages will structure their sentences differently, where the {term} and {category} might be mid-sentence. I don't have any concrete examples but I do remember running into this problem the last time I worked on a localized app (at BV).

Reply:
More interesting might be the idea of two separate translations, one for "Showing results for {term}" and another for "Showing results for {term} in {category}".

Then the component would decide which _**`id`**_ to use based on the truthiness of `category`.

Reply:
> should in be highlighted?

Ooh, I like it! https://github.com/magento/pwa-studio/pull/2685/commits/ccf898cc9e403981db502b41a747c39b2b0dd3a5